### PR TITLE
Add test for var local-func return type

### LINKED ISSUE #10392: Local functions: Remove support for `var` return type
May possibly be relaxed later.

### PR DESCRIPTION
**Pull request by agocke**
Support for using `var` in a local function return type was already
removed in PR #7916. This PR adds a test to verify that.

Closes #10392.

@jaredpar @VSadov @gafter  @dotnet/roslyn-compiler Simple change, please review.

### CODE REVIEW

**Comment by agocke** [comment]
test prtest/lin/dbg/unit32 please

File: src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs [hunk 1]
```diff
@@ -77,6 +77,24 @@ private void VerifyDiagnostics(string source, CSharpCompilationOptions options,
...
+        int x = local();
+   } 
+}";
+            VerifyDiagnostics(src,
+                // (6,9): error CS0825: The contextual keyword 'var' may only appear within a local variable declaration or in script code
+                //         var local() => 0;
+                Diagnostic(ErrorCode.ERR_TypeVarNotFound, "var").WithLocation(6, 9));
+        }
+
+        [Fact]
```
Comment (jaredpar) [inline src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs]: Need to add the positive test: case where `var` is a valid user defined type.  This should still work.

**Comment by jaredpar** [comment]
LGTM pending adding a positive test here.

File: src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs (hunk 1 above)
Comment (jaredpar) [inline src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs]: Also what hapens if `var` is an identifier in the function.  What error do we produce?

**Comment by agocke** [comment]
test prtest/lin/dbg/unit32 please

**Comment by gafter** [comment]
LGTM

**Comment by agocke** [comment]
Closing and re-opening against `master`

**Pull request closed without merging**